sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs

```diff
@@ -1999,6 +2008,22 @@ public partial class SearchSuggester
         public string Name { get { throw null; } set { } }
         public System.Collections.Generic.IList<string> SourceFields { get { throw null; } }
     }
+    public partial class SemanticConfiguration
+    {
+        public SemanticConfiguration(string name, Azure.Search.Documents.Indexes.Models.PrioritizedFields prioritizedFields) { }
+        public string Name { get { throw null; } set { } }
+        public Azure.Search.Documents.Indexes.Models.PrioritizedFields PrioritizedFields { get { throw null; } set { } }
+    }
+    public partial class SemanticField
+    {
+        public SemanticField() { }
+        public string FieldName { get { throw null; } set { } }
+    }
+    public partial class SemanticSettings
+    {
+        public SemanticSettings() { }
+        public System.Collections.Generic.IList<Azure.Search.Documents.Indexes.Models.SemanticConfiguration> Configurations { get { throw null; } }
+    }
```

Comment:
Does the service expect these classes to grow?  Right now it feels like _a lot_ of empty types that don't add much value.  Today I have to write:

```C#
index.SemanticSettings =
    new SemanticSettings
    {
        Configurations = {
            new SemanticConfiguration
            {
                Name = "Whatever",
                PrioritizedFields = new PrioritizedFields
                {
                    TitleField = new SemanticField { FieldName = "Title" },
                    ContentFields = {
                        new SemanticField { FieldName = "Description" },
                        new SemanticField { FieldName = "Summary" }
                    },
                    KeywordsFields  = {
                        new SemanticField { FieldName = "Title" },
                        new SemanticField { FieldName = "Summary" }
                    }
                }
            }
        }
    };
```
Here's what the same information could look like if we dropped all the types:
```C#
index.SemanticConfigurations["Whatever"] =           // Dictionary<string, SemanticConfiguration>
    new SemanticConfiguration(
        title: "Title",                               // string
        content: new[] { "Description",  "Summary" }, // IEnumerable<string>
        keyword: new[] { "Title",  "Summary" });
```

Reply:
The idea was to be able to support other type of schemas (something that would replace "PrioritizedFields" and offer a different set of expected fields). Also, the "Fields" themselves are an object rather than just a string containing field names to keep the flexibility to add additional parameters to individual fields (such as truncation type). We don't necessarily have short term plans to leverage those new richer types, but we wanted to design an API that allow us to add them easily.

Reply:
@Mohit-Chakraborty - let's keep the current API as-is right now, but consider adding a .ctor like `public SemanticConfiguration(string name, string titleField = null, IEnumerable<string> contentFields = null, keywordFields = null)` in a future release.